src/debugAdapter/goDebug.ts

```diff
@@ -835,10 +835,13 @@ class GoDebugSession extends LoggingDebugSession {
 		if (this.delve.remotePath.length === 0) {
 			return this.convertClientPathToDebugger(filePath);
 		}
+		// When the filePath has a different path separator
+		// than the local path separator (cross-compilation),
+		// the split and join logic won't work.
+		// See github.com/microsoft/vscode-go/issues/2010.
```

Comment:
The below manipulation is also required when the user provided `program` uses a different path separator than what VS Code passes to the debug adapter.

```suggestion
		// The filePath may have a different path separator than the localPath
		// So, update it to use the same separator as the remote path to ease
		// in replacing the local path with remote path
```